packages/webapp/components/layouts/ProfileLayout/index.tsx

```diff
@@ -92,6 +93,7 @@ export default function ProfileLayout({
 
   // Needed because sometimes initialProfile is defined and fetchedProfile is not
   const profile = fetchedProfile ?? initialProfile;
+  const isCurrentUserProfile = profile && profile.id === user?.id;
```

Comment:
To continue on the comment https://github.com/dailydotdev/apps/pull/2343/files#r1389695102

Basically here you can call `const { isReady: isReferralReady } = useReferralCampaign` and then before below add `isReferralReady` check and return null until data is ready.

Reply:
Per DR this will delay render of profile but it will avoid layout shift.

Reply:
Also we already have this evaluation on the `pages/userId/index` file as:
```
const isSameUser = profile?.id === user?.id;
```

Not sure if you can re-use it?

Reply:
We can make some util fn that checks both

Reply:
Agreed to keep this as is